### DISCUSSION TITLE
Convert f090 Mining Reserve actor to a keyless account actor

### DISCUSSION THREAD

**Discussion by jennijuju** [filecoin-project/FIPs#901] [General]
> FIP Draft: https://github.com/filecoin-project/FIPs/pull/903

idea from chitchat with @jnthnvctr 

## Abstract

Covert f090 (FIL mining reserve) actor type from a _multisig_ to an _account_, and further transfer the ownership, control and governance of the funding reserved from the signers of the multisig back to the network's participants via community governance. 

## Motivation

As documented in the spec [here](https://spec.filecoin.io/#section-systems.filecoin_token.token_allocation), upon network launch, a pool of tokens was allocated as a Mining Reserve to support the maintenance of a robust Filecoin economy. It also mentions, "It will be up to the community to determine in the future how to distribute those tokens, through Filecoin improvement proposals (FIPs) or similar decentralized decision-making processes." It is worth noting that when the spec was written and when the network was first launched, the FIP process was not yet well-designed or matured to support network governance, and significant improvements have been made since then.

At the moment, the actor holding the FIL mining reserve is `f090`, which is a multisig actor with 3 signers and an approval threshold of 2. All three signers are account actors, with wallet keys to those accounts. This means three individuals hold the keys to three wallets and can move funds and change the terms of f090, as long as 2 out of 3 signers approve the same transactions. They can do so without anyone else's approval, protocol-wise speaking. This opposes the decentralized governance principle as quoted in the above paragraph from the spec. This also creates a security black hole in the network where if 2 out of the 3 signer keys of f090 are compromised by malicious actors, they may cause serious economic damage to the Filecoin network.

In addition, having the reserve holds in a multisig also means changes toward the mining reserve, including unlock portion of the funds to new network participants or any operation/mechanism accepted by the community via FIP, may need to be managed by the msig signers via signed transactions which is a huge operational overhead (and again against the principle).

The proposal here is rather simple: to convert the f090 actor type from a *multisig* to an *account* type (without any keys, like `f099`), and any changes to the mining reserve must be proposed via FIPs, governed by the FIP and network upgrade process.

## Security Consideration

This proposal improves the network security by removing the ownership and control of 15% of the total network token supply from 3 individuals to the network participants. 

Unless there is any downsides of this proposal is being oversaw (if so, please don't hesitate to comment below!), otherwise the implementation of this proposal should be trivial and ideally, could be implemented in an upcoming network upgrade.

**Comment by rjan90** [comment]
I agree with the proposal!

While there might have been reasons for launching the mining reserve as a multisig actor at the network's inception. Converting the f090 actor from a multisig to an account type without any keys, and any changes must be proposed via FIPs and a network upgrade process aligns more closely with the principles of decentralized governance.

**Comment by deltazxm** [comment]
Suggest burning this part directly to reduce inflation

**Reply by jennijuju** [reply]
Such CE change should be its own FIP, and I believe this FIP is a prerequisite for any potential changes to be proposed wrt mining reserve.

**Comment by steven004** [comment]
Outstanding insights, @jennijuju. I support your viewpoint.

Simultaneously, I believe directly burning the tokens is a sound strategy, as @deltazxm mentioned. Initially, at Filecoin's launch, reserving a portion of tokens for future incentives made sense despite the lack of a concrete plan for their use. This was primarily due to our reliance on native tokens without a roadmap for supporting secondary tokens via a virtual machine. Now, as incentivization can be effectively managed through smart contract economics, the necessity of these reserves is no longer as critical.

**Comment by rvagg** [comment]
Discussion of burning is probably going to complicate this discrete proposal and maybe should be raised in a separate proposal? I don't think it either blocks or necessarily conflicts with this proposal but bundling in such a discussion here might make it harder to take the simple steps proposed and is likely to be more contentious.

**Comment by jennijuju** [comment]
given the feedback received so far (here and in DMs), it seems like the proposal is not too controversial, so I went ahead and opened an FIP Draft https://github.com/filecoin-project/FIPs/pull/903.

Im hoping to get more peer reviews to confirm that proposed solution is simple to implement. IIUC it should just be an actor type migration, but would be good to be sure.

**Comment by anorth** [comment]
This proposal sounds great to me. The multisig probably made some sense at the time of network launch, but doesn't any more.

FWIW I think burning the tokens would be quite short-sighted. The reserve balance contributes nothing to inflation today (or when locked in a keyless account) because it's are a static stock of tokens, not an inflow to supply. The undisbursed amount is not part of the calculated circulating supply (and this will be more obviously correct after this proposal). These tokens represent an option, and I can't see much value in destroying that option. (But it's for sure worth discussing as a separate, subsequent proposal).

**Reply by nai2283** [reply]
It's quite surprising that you call burning the reserve "short sighted", it's locked today doesn't mean it will be locked forever. (And if so there would be no difference as burning), but existence of the reserve itself does create certain level of uncertainty. Plus we all know PL (at least some members from it) is not famous because they respect "code is law", so many people will feel more comfortable to just burn it. With that being said this FIP is a great step forward and burning or not shall be a separate topic.

**Reply by anorth** [reply]
I changed my mind here: https://github.com/filecoin-project/FIPs/discussions/1030#discussioncomment-9930704

Eagle-eye @jennijuju pointed this out, so I thought I'd connect the dots.

**Comment by Fatman13** [comment]
why not locking the 2.5% fund raising remainder too?

**Comment by zixuanzh** [comment]
Sounds like everyone who participated in this thread so far are in broad agreement, any progress / update on this FIP?

**Reply by rjan90** [reply]
The FIP has been accepted: https://github.com/filecoin-project/FIPs/blob/master/FIPS/fip-0085.md.

It has also been implemented and tested in a devnet: https://github.com/filecoin-project/lotus/issues/11939#issuecomment-2099865595. Scheduled to go out with nv23.

**Reply by zixuanzh** [reply]
Great! @jnthnvctr @tmellan I think there's a lot that can be done to market this
